types/backbone/backbone-tests.ts

```diff
@@ -345,6 +346,10 @@ function test_collection() {
     one = books.sample();
     models = books.sample(3);
     models = books.select((value: Book, index: number, list: Book[]) => true);
+    oneOrMany = books.where({name: 'Mike'});
```

Comment:
Could you check type with `// $ExpectType` please, and also could you add a error case when first not boolean?

Reply:
I got this when `first` omit or undefined.

```
oneOrMany = books.where({name: 'Mike'}); // $ExpectType Book | Book[]
```

Is there way can infer the type to be `Book[]` cause `undefined` isn't true.

Reply:
I looked more closely, I have several reasons for leaving the code as it is:

1. The documentation https://backbonejs.org/#Collection-where does not describe such parameter, but in the source code it is there. I think the author of Backbone.js just wanted to reuse the function for `findWhere` and didn't write it quite cleanly.

2. Using `books.findWhere({name: 'Mike'})` seems easier and more readable than `books.where({name: 'Mike'}, true)`.

3. We would not break the current code that other developers are already using in case when parameter not passed.

Reply:
Good point!
Well my changes do seem break current code.
Before: it return `Book[]`
After: `Book[] | Book`

Not sure if overloads works in terms of backward compatibility. 
```
        where(properties: any): TModel[];
        where<F extends boolean>(properties: any, first?: F): F extends true ? TModel : TModel[];
```

Reply:
and TBH I don't why it infers `TModel[] | TModel` when `first` is undefined, which is clearly not `extends true`?
```
oneOrMany = books.where({name: 'Mike'}, undefined); // $ExpectType Book | Book[]
```